R/translator.R

```diff
@@ -0,0 +1,108 @@
+#' Translator options
+.translator_options <- list(
+  cultural_bignumer_mark = NULL,
+  cultural_punctuation_mark = NULL,
+  cultural_date_format = NULL
+)
+
+#' Translator class
+#'
+#' @field languages character.
+#' @field key_translation character.
+#' @field translation_file character.
+#' @field options list.
+#' @field translations data.frame.
+#' @field translation_language character.
+#'
+#' @return
+#' @export
+#'
+#' @import jsonlite
+#'
+#' @examples
+#' @rdname translator
+Translator <- setRefClass(
+  "Translator",
+  fields = list(
+    languages = "character",
+    translation_language = "character",
+    options = "list",
+    translations = "data.frame",
+    mode = "character"
+  )
+)
+
+#' @rdname translator
+Translator$methods(
+  initialize = function(translation_csvs_path = NULL,
+                        translation_json_path = NULL,
+                        translation_csv_config = NULL) {
+    options <<- .translator_options
+    # Reading translations.
```

Comment:
I'd say this comment is unnecessary, but perhaps it's ok ;) I highly recomment the chapter *Self-Documenting Code* from the book *Code Complete* -- it has a lot of useful advice about when to write comments